runtime/compiler/env/J9KnownObjectTable.cpp

```diff
@@ -115,13 +117,22 @@ J9::KnownObjectTable::getOrCreateIndex(uintptr_t objectPointer)
 
 
 TR::KnownObjectTable::Index
-J9::KnownObjectTable::getOrCreateIndex(uintptr_t objectPointer, bool isArrayWithConstantElements)
+J9::KnownObjectTable::getOrCreateIndex(uintptr_t objectPointer, bool isArrayWithConstantElements, int32_t stableArrayRank)
    {
    TR::KnownObjectTable::Index index = self()->getOrCreateIndex(objectPointer);
    if (isArrayWithConstantElements)
       {
       self()->addArrayWithConstantElements(index);
       }
+
+   if (stableArrayRank > 0)
```

Comment:
I can't find any check whether the object is an array for addArrayWithConstantElements. Do you mean we should check for both?

Reply:
I was only commenting about stable arrays, since those are what's new, but I do think it would make sense for `addArrayWithConstantElements()` as well

Reply:
I added assert to only addStableArray to minimize the changes in this PR.